### PR TITLE
Basic return arguments + Visit function

### PR DESCRIPTION
**Pull request by zimmski**

### CODE REVIEW

File: cmd/go-clang-generate/type.go [hunk 1]
```diff
@@ -67,45 +71,58 @@ func getType(cType clang.Type) (Type, error) {
 	switch cType.Kind() {
 	case clang.TK_Char_S:
 		typ.CName = CSChar
-		typ.Name = string(GoInt8)
+		typ.Name = GoInt8
+		typ.Primitive = "schar"
```
Comment (marxriedl) [inline cmd/go-clang-generate/type.go]: you are aware that every typ.Primitive is identical with typ.CName?

Comment (marxriedl) [inline cmd/go-clang-generate/type.go]: and that i have defined constants for that ^^

File: cmd/go-clang-generate/main.go [hunk 2]
```diff
@@ -331,39 +383,62 @@ func main() {
...
+				p.Type.Primitive = e.Receiver.Type.Primitive
 			} else if _, ok := lookupStruct[p.Type.Name]; ok {
-			} else {
-				if goType, primitiveType := goAndTypePrimitive(p.Type.Name); goType != "" {
-					p.Type.Name = goType
-					p.Type.Primitive = primitiveType
+			}
+
+			// TODO happy hack, whiteflag types that are return arguments
+			if p.Type.PointerLevel == 1 && (p.Type.Name == "File" || p.Type.Name == "FileUniqueID" || p.Type.Name == "IdxClientFile" || p.Type.Name == "cxstring" || p.Type.Name == GoUInt16) {
```
Comment (marxriedl) [inline cmd/go-clang-generate/main.go]: possibly define a whitelist map? would get rid of this || p.Type.Name == cascade

**Comment by marxriedl** [comment]
seems mostly fine; you can decide if you don't want to address the comments, you can merge immediately

File: cmd/go-clang-generate/type.go (hunk 1 above)
Comment (zimmski) [inline cmd/go-clang-generate/type.go]: OK so "CName" is not really the C-name but the CGo name? I will correct that and rename some things.

Comment (zimmski) [inline cmd/go-clang-generate/type.go]: Done

File: cmd/go-clang-generate/main.go (hunk 2 above)
Comment (zimmski) [inline cmd/go-clang-generate/main.go]: This is just a workaround for now, I want to get rid of this.

**Pull request merged**